charts/icm-as/templates/as-deployment.yaml

```diff
@@ -61,10 +61,30 @@ spec:
             value: "false"
           - name: SERVER_NAME
             value: "appserver"
-          {{- if .Values.debug.enabled }}
+          {{- if .Values.jvm.debug.enabled }}
           - name: DEBUG_ICM
             value: "true"
           {{- end }}
+          {{- if .Values.jvm.heap.min }}
+          - name: XMS
+            value: {{ printf "%s%s" "-Xms" .Values.jvm.heap.min | quote }}
+          {{- end }}
+          {{- if .Values.jvm.heap.max }}
+          - name: XMX
+            value: {{ printf "%s%s" "-Xmx" .Values.jvm.heap.max | quote }}
+          {{- end }}
```

Comment:
Why you don't use -XX:+UseContainerSupport? With this setting, heap size can be calculated automatically based on the given resources. Currently the user has to set the resources and after that to calculate the Java memory settings manually. This is a potential source of failures.

Reply:
UseContainerSupport is there by default (since Java 10) and allows java to see the container limits and not the host ones.
The one you are talking about is -XX:MaxRAMFraction. But normally you don't want to let the container decide how big your application is. It's like your clothes decide about your figure.
Actual best practices stay with setting xmx. But it's also recommended to set xms equal to xmx to spare gc pause times which occurs everytime xms has to be enhanced.